src/Api/Controllers/CiphersController.cs

```diff
@@ -360,6 +368,83 @@ public async Task DeleteMany([FromBody]CipherBulkDeleteRequestModel model)
             await _cipherService.DeleteManyAsync(model.Ids.Select(i => new Guid(i)), userId);
         }
 
+        [HttpPut("{id}/delete")]
+        public async Task PutDelete(string id)
+        {
+            var userId = _userService.GetProperUserId(User).Value;
+            var cipher = await _cipherRepository.GetByIdAsync(new Guid(id), userId);
+            if (cipher == null)
+            {
+                throw new NotFoundException();
+            }
+            await _cipherService.SoftDeleteAsync(cipher, userId);
```

Comment:
Did you consider just using existing update methods? Then you wouldn't need all these extra services, repos, and sprocs.

Reply:
We did consider that, leaving all of the logic to the client, however in that same conversion I'm pretty sure we landed on making soft-delete and restore explicit actions called by the client against the API (similar to deleting or updating today). I can certainly dial back in the DB side of things to just use Update and remove all of the other sprocs and repository methods if you would prefer that.

Reply:
I meant using update services and things that the server already provides. Not doing it on the client. For example here, your soft delete service could just set `RevisionDate = DeletedDate = DateTime.UtcNow` and call the replace responsitory.